networkx/algorithms/tests/test_swap.py

```diff
@@ -123,3 +123,18 @@ def test_degree_seq_c4():
     degrees = sorted(d for n, d in G.degree())
     G = nx.double_edge_swap(G, 1, 100)
     assert degrees == sorted(d for n, d in G.degree())
+
+
+def test_no_edges():
+    G = nx.DiGraph()
+    G.add_nodes_from([0, 1, 2])
+    pytest.raises(nx.NetworkXError, nx.directed_edge_swap, G)
+    G = nx.Graph()
+    G.add_nodes_from([0, 1, 2, 3])
+    pytest.raises(nx.NetworkXError, nx.double_edge_swap, G)
+
+
+def test_less_than_3_edges():
+    G = nx.DiGraph()
+    G.add_edges_from([(0, 1), (1, 2)])
+    pytest.raises(nx.NetworkXError, nx.directed_edge_swap, G)
```

Comment:
```suggestion
    G = nx.DiGraph([(0, 1), (1, 2)])
    with pytest.raises(nx.NetworkXError, match=".*fewer than 3 edges"):
        nx.directed_edge_swap(G)
```

Note that the below wording for `match=` is predicated on the assumption that we update 'less' -> 'fewer' in the exception message.